crates/test/tracing-subscriber/src/lib.rs

```diff
@@ -15,14 +15,21 @@ pub fn set_tracing_subscriber() -> DefaultGuard {
     let filter = "info,penumbra_app=trace,penumbra_mock_consensus=trace";
     let filter = EnvFilter::try_from_default_env()
         .or_else(|_| EnvFilter::try_new(filter))
-        .expect("should have a valid filter directive")
-        // Without explicitly disabling the `r1cs` target, the ZK proof implementations
-        // will spend an enormous amount of CPU and memory building useless tracing output.
-        .add_directive(
-            "r1cs=off"
-                .parse()
-                .expect("rics=off is a valid filter directive"),
-        );
+        .expect("should have a valid filter directive");
+    set_tracing_subscriber_with_env_filter(filter)
+}
+
+/// Install a tracing subscriber with a custom filter.
+//  NB: this is marked as "dead code" but it is used by integration tests.
+#[allow(dead_code)]
+pub fn set_tracing_subscriber_with_env_filter(filter: EnvFilter) -> DefaultGuard {
```

Comment:
This adds a method to specify a custom `EnvFilter`. I'm not totally sure this is the right api, or desired. I added this so that I could scope the tracing logs to the exact subset that's relevant to observe the bug that this test repros, cc @cratelyn

Reply:
yes yes yes!! very much in favor of doing this, i'd pondered doing before as well. :pray: